src/adapter/src/coord.rs

```diff
@@ -1903,6 +1903,9 @@ impl Coordinator {
 
             let (idle_tx, mut idle_rx) = tokio::sync::mpsc::channel(1);
             let idle_metric = self.metrics.queue_busy_seconds.with_label_values(&[]);
+            let last_message_kind = Arc::new(Mutex::new("none"));
+            let last_message_kind_ = Arc::clone(&last_message_kind);
```

Comment:
Are these both supposed to be here? This looks weird to me.

Reply:
They are both supposed to be here, we need one to move into the "coord watchdog" task, and the other for the coord loop. I moved things around a little bit, let me know what you think